subprojects/core/src/main/java/org/gradle/api/services/internal/BuildServiceDetails.java

```diff
@@ -25,6 +25,7 @@
  * Details about a service registration.
  */
 public class BuildServiceDetails<T extends BuildService<P>, P extends BuildServiceParameters> {
+    public static final int DEFAULT_MAX_USAGES = -1;
```

Comment:
💅 does it have to be public?

Reply:
There are two use cases for `-1` right now: (1) being a default, (2) being a special-case marker for unlimited usage. It'd be nice to make it clear via the declaration rather than the usage sites.

I'd go for extra explicit and have two variables:
```
MAX_USAGES_UNLIMITED = -1
DEFAULT_MAX_USAGES = MAX_USAGES_UNLIMITED
```

For the two cases we'd actually use the different variables.

If that's too verbose for your taste, then I'd rename the current variable to `MAX_USAGES_UNLIMITED`. It being a default could be inferred from the usage in the constructor. But it's less explicit

Reply:
🙏 💅   Can we also add an empty line after the class declaration?